### PR TITLE
Added block header

### PR DESCRIPTION
**Pull request by dgarcia360**
Note: The order of the parameters could be wrong.

### CODE REVIEW

**Comment by dgarcia360** [comment]
@Jaguar0625 Resolved comments

File: schemas/block.cats [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	inline EntityBody
+
+	# block height
+	height = uint64
+
+	# number of seconds elapsed since creation of nemesis block
+	timestamp = Timestamp
+
+	# block difficulty
+	difficulty = uint64
```
Comment (Jaguar0625) [inline schemas/block.cats]: add Difficulty type

File: schemas/block.cats [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	# previous block hash
+	previousBlockHash = Hash256
+
+	# block transactions root hash
+	transactionHash = Hash256
+
+	# block receipts root hash
+	receiptsHash = Hash256
+
+	# block states root hash
```
Comment (Jaguar0625) [inline schemas/block.cats]: Hash of the global chain state at this block ?

File: schemas/block.cats [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	timestamp = Timestamp
+
+	# block difficulty
+	difficulty = uint64
+
+	# previous block hash
+	previousBlockHash = Hash256
+
+	# block transactions root hash
+	transactionHash = Hash256
```
Comment (Jaguar0625) [inline schemas/block.cats]: transaction*s

File: schemas/block.cats [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import "entity.cats"
+
+# binary layout for a block header
+struct BlockHeader
+	inline VerifiableEntity
```
[the remark this answers is not in the document]
Comment (dgarcia360) [reply schemas/block.cats]: Added, please review if it is defined in the proper line (first inline).

**Pull request merged**